### ISSUE TITLE
Unable to load DLL 'libmagic-1' or one of its dependencies(HRESULT: 0x8007007E)

### ISSUE THREAD

**Issue by Serfi7** [hey-red/Mime#25]
I have an app based on ASP.NET Core 2.1. At localhost all works fine. But than I published my app I got the problem described at title. At my publication folder I found "labmagic-1.dll" at "\runtimes\win-x64\native". I set "MimeGuesser.MagicFilePath" at absolute and relative path. But didn't help.

**Comment by hey-red** [comment]
You are using self-contained deployment? You should place all files from \runtimes\win-x64\native into output directory.

**Comment by Serfi7** [comment]
You mean place it to root of app? If I place files to root I have next problem: 

> BadImageFormatException: An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)

**Comment by hey-red** [comment]
Hmm interesting. You are publish through visual studio or command line?
What OS? Is x64 windows?
>Am I need install net core sdk 2.0 to my server?
You can install only .net core runtime or full sdk(the runtime is included).
Also, see https://docs.microsoft.com/en-us/dotnet/core/deploying/index

**Comment by Serfi7** [comment]
Yes, I'm using VS and Web Deploy to publish to my windows server 2016(x64).

**Comment by hey-red** [comment]
Ok. You can show your "publish" directory? Its contains "runtimes" dir?

**Comment by Serfi7** [comment]
Yes, it contains "runtimes" and "libmagic-1.dll" next.

![default](https://user-images.githubusercontent.com/37456151/43353694-119bb0d4-9247-11e8-85bc-b9937db6d28e.png)

![default](https://user-images.githubusercontent.com/37456151/43353711-464b9272-9247-11e8-974b-4d609eb521e3.png)

**Comment by cguillencr** [comment]
When I published my project using dotnet cli
 ` sudo dotnet publish AA.csproj -c Release -r ubuntu.14.04-x64`

The file magic.mgc  is been created inside the published folder, but for some reazon only linux was not able to see it.  Thats why I place this code in the startup of my mvc project .

```
    if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                MimeGuesser.MagicFilePath = "magic.mgc";
            }
```
Run 
`find / -name magic.mgc` to verify the .mgc file is inside the proper folder you may use the absolute path as well.

Watch out if you face this error
`MagicException: File 5.30 supports only version 14 magic files. `/usr/share/file/magic.mgc' is version 10` Means that I must upgrade linux o downgrade  Mime version in your Nuget.

I my case the project was published and tested in ubuntu18
 ` sudo dotnet publish AA.csproj -c Release -r ubuntu.18.04-x64`
